opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -55,5 +54,23 @@
    * (Note that some OpenTracing implementations consider the Spans on either side of an RPC to have the same identity,
    * and others consider the caller to be the parent and the receiver to be the child)
    */
-  <T> Span join(String operationName, T carrier);
+  <T> SpanBuilder join(String operationName, T carrier);
```

Comment:
If it is to be @Nullable then i'd rather see it go into the builder, ie `withOperationName(..)`.

The builder is entitled to document which methods are expected to be called (even in different combinations) and for `start()` to throw an AssertionError if still unset when it should be.